shared/tsconfig.base.json

```diff
@@ -1,9 +1,8 @@
 {
   "compilerOptions": {
-    "outDir": "${configDir}/build",
-    "noEmit": true,
+    "outDir": "${configDir}/node_modules/.cache/.tsc_cache/build",
```

Comment:
All projects now emit to a cache directory, which the TypeScript compiler will use (see [Project References](https://www.typescriptlang.org/docs/handbook/project-references.html)). We had previously only set this up for a subset of projects.

Our projects that do not use a bundler still emit to `./dist`.